mypy/semanal.py

```diff
@@ -1659,7 +1659,7 @@ def visit_import_all(self, i: ImportAll) -> None:
                 # False, and we can skip checking '_' because it's been explicitly included.
                 if (new_node and new_node.module_public and
                         (not name.startswith('_') or '__all__' in m.names)):
-                    existing_symbol = self.globals.get(name)
+                    existing_symbol = self.lookup_current_scope(name)
```

Comment:
What's the story with this? Was there an existing bug here that just got exposed by one of these test cases, or is this only necessary with fine-grained mode?

Reply:
I believe that there was an existing bug, but it could be a bigger problem in fine-grained incremental mode.